File: llt.py
mersenne_primes = [6, 28, 496, 8128, 33550336, 8589869056, 137438691328]

def llt(p, found):
    s = 4
    M = 2**p - 1

    for index in range(p - 2):
        print(f"\revaluating {p} as candidate for perfect number - {round(index / (p - 2) * 100, 6)}% completed", end="")
    
        s = (s * s - 2) % M

    print("\n")

    for f in found:
        print(f"{f} found to be a prime")

    return s == 0

found = []
for m in mersenne_primes: # Takes 1 minute to process 0.0001% of 5th marsenne prime
    found.append(m)
    print(f"{m} is {'prime' if llt(m, found) else 'composite'}")

File: llt.c
#include <stdio.h>  
#include <stdlib.h> 
#include <math.h>   
#include <stdbool.h>

int llt(long long int p, long long int found[], int size) {
    long long int s = 4;
    long long int M = (1LL << p) - 1; // M = 2^p - 1

    for (int index = 0; index < p - 2; index++) {
        printf("\revaluating %lld as candidate for perfect number - %.2f%% completed", p, (float)index / (p - 2) * 100);
        fflush(stdout);

        s = (s * s - 2) % M;
    }
    printf("\n");

    for (int i = 0; i < size; i++) {
        printf("%lld found to be a prime\n", found[i]);
    }

    return s == 0;
}

int main() {
    long long int mersenne_primes[10] = {6, 28, 496, 8128, 33550336, 8589869056, 137438691328}; // 3% a second on 5th marsenne prime
    long long int found[10]; 
    int p;

    for (int i = 0; i < sizeof(mersenne_primes) / sizeof(mersenne_primes[0]); i++) {
        p = mersenne_primes[i];
        found[i] = p;
        printf("%lld is %s\n", p, llt(p, found, i + 1) ? "prime" : "composite");
    }

    return 0;
}